### PR TITLE
Wdb 61 configure authentication in backend

### PR DESCRIPTION
**Pull request by salmonela7**
I don't know if this is the best implementation. Feel free to suggest a better way to do this :D

### CODE REVIEW

File: backend/backend/JwtAuthentication/JwtAuthentication.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        public JwtAuthentication(string tokenKey, BackendContext context)
+        {
+            _tokenKey = tokenKey;
+            _db = context;
+        }
+
+        public string Authenticate(string email, string password)
+        {
+            if (!_db.Users.Any(u => u.Password == password && u.Email == email))
```
Comment (kibartas) [inline backend/backend/JwtAuthentication/JwtAuthentication.cs]: The password should be hashed. Hashing will probably be implemented alongside 'Register' though, so maybe we'll just add it later here.

File: backend/backend/Controllers/UsersController.cs [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
+using backend.Models;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace backend.Controllers
 {
     [Route("api/[controller]")]
+    [Authorize]
```
Comment (kibartas) [inline backend/backend/Controllers/UsersController.cs:13]: could you explain this annotation?

**Review by kibartas — APPROVED** [review]
I haven't done JWT in C# so I'm not sure if everything is done 100% correctly but LGTM

File: backend/backend/JwtAuthentication/JwtAuthentication.cs (hunk 1 above)
Comment (salmonela7) [reply backend/backend/JwtAuthentication/JwtAuthentication.cs]: We have another task for password encryption. This is just a "sample" implementation of authentication

File: backend/backend/Controllers/UsersController.cs (hunk 2 above)
Comment (salmonela7) [reply backend/backend/Controllers/UsersController.cs:13]: Means that only authenticated users can access these endpoint

Except the authentication endpoint as it has AllowAnonymous attribute

File: backend/backend/JwtAuthentication/JwtAuthentication.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(_tokenKey);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new[]
+                {
+                    new Claim(ClaimTypes.Name, email)
+                }),
+                Expires = DateTime.UtcNow.AddHours(1),
```
Comment (AArlauskas) [inline backend/backend/JwtAuthentication/JwtAuthentication.cs:38]: Any ideas for implementing refresh tokens?

File: backend/backend/JwtAuthentication/JwtAuthentication.cs (hunk 1 above)
Comment (AArlauskas) [inline backend/backend/JwtAuthentication/JwtAuthentication.cs]: Maybe you should change "Any" to "First" and then use user's GUID for Claims

File: backend/backend/JwtAuthentication/JwtAuthentication.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                return null;
+            }
+
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(_tokenKey);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new[]
+                {
+                    new Claim(ClaimTypes.Name, email)
```
Comment (AArlauskas) [inline backend/backend/JwtAuthentication/JwtAuthentication.cs]: This should contain GUID for user identification

File: backend/backend/Controllers/UsersController.cs [hunk 5]
```diff
@@ -1,25 +1,45 @@
...
+            _jwtAuthentication = jwtAuthentication;
         }
 
         [HttpGet,Route("all")]
         public ActionResult<List<User>> GetAllUsers()
         {
-            return db.Users.ToList();
+            return _db.Users.ToList();
+        }
+
```
Comment (AArlauskas) [inline backend/backend/Controllers/UsersController.cs:33]: Would be nice to see an example of how you identify the user and fetch his data

**Review by AArlauskas — CHANGES_REQUESTED** [review]
Overall, looks very good. Nice job. Just a few things need fixing

File: backend/backend/JwtAuthentication/JwtAuthentication.cs (hunk 3 above)
Comment (salmonela7) [reply backend/backend/JwtAuthentication/JwtAuthentication.cs:38]: No idea currently, maybe I will investigate more. Should we create a task for it in backlog?

File: backend/backend/JwtAuthentication/JwtAuthentication.cs (hunk 1 above)
Comment (salmonela7) [reply backend/backend/JwtAuthentication/JwtAuthentication.cs]: Done

File: backend/backend/Controllers/UsersController.cs (hunk 5 above)
Comment (salmonela7) [reply backend/backend/Controllers/UsersController.cs:33]: Done

File: backend/backend/JwtAuthentication/JwtAuthentication.cs (hunk 4 above)
Comment (salmonela7) [reply backend/backend/JwtAuthentication/JwtAuthentication.cs]: Done

**Review by AArlauskas — APPROVED** [review]
Everything looks good

**Review by povilassta — APPROVED** [review]
Nice work

**Pull request merged**